### PR TITLE
Implement brewdle upgrade command.

### PR DESCRIPTION
**Pull request by AlJohri**
This PR is a work in progress. See discussion on this issue: https://github.com/Homebrew/homebrew-brewdler/issues/37.

### CODE REVIEW

File: bin/brewdle [hunk 1]
```diff
@@ -20,3 +20,12 @@ command :install do |c|
...
 end
+
+command :upgrade do |c|
+  c.description = 'Install and Upgrade all Homebrew based dependencies'
+  c.action do |args, options|
+    Brewdler::Commands::Upgrade.run
+  end
+end
+
+alias_command :'update', :'upgrade'
```
Comment (mikemcquaid) [inline bin/brewdle]: This should probably be the default rather than a separate command?

File: lib/brewdler.rb [hunk 2]
```diff
@@ -3,4 +3,6 @@
 require "brewdler/brew_installer"
 require "brewdler/cask_installer"
 require "brewdler/repo_installer"
+require "brewdler/commands/command"
 require "brewdler/commands/install"
+require "brewdler/commands/upgrade"
```
Comment (mikemcquaid) [inline lib/brewdler.rb]: Missing a newline at the end here.

File: lib/brewdler/dsl.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module Brewdler
   class Dsl
-    def initialize(input)
+    def initialize(input, mode)
```
Comment (mikemcquaid) [inline lib/brewdler/dsl.rb]: Might want to use a `options={}` hash here instead and pass `:mode => :install` to make it a bit more Ruby-like (and handle the case where we add more args later)

File: spec/dsl_spec.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 describe Brewdler::Dsl do
-  let(:dsl) { Brewdler::Dsl.new("tap 'phinze/cask'\nbrew 'git'\ncask 'google-chrome'\nbrew 'emacs', args: ['cocoa', 'srgb', 'with-gnutls']") }
+  let(:dsl) { Brewdler::Dsl.new("tap 'phinze/cask'\nbrew 'git'\ncask 'google-chrome'\nbrew 'emacs', args: ['cocoa', 'srgb', 'with-gnutls']", "install") }
```
Comment (mikemcquaid) [inline spec/dsl_spec.rb]: I think `install` should remain as the default.

Comment (AlJohri) [inline spec/dsl_spec.rb]: sounds good. I'll make these changes.

**Comment by andrew** [comment]
FYI some of this is going to conflict with https://github.com/Homebrew/homebrew-brewdler/pull/39

**Comment by andrew** [comment]
@AlJohri #39 has been merged now, can you rebase against master and fix the conflicts?

**Comment by andrew** [comment]
@AlJohri any progress on this?

**Comment by mikemcquaid** [comment]
Closing this due to no activity. Feel free to reopen a new PR later!

**Pull request closed without merging**